### ISSUE TITLE
Feature: loop for patter calls: Define maximum number of loops 

### ISSUE THREAD

**Issue by Gero5** [mpogue2/SquareDesk#1209]
**current situation:**
For a patter call when loop is active it is an infinite loop. Therefore patter calls are getting too long (in my case). It also hinders me to end the patter call with the actual closing part, i. e. the last beat of the patter music. 

**New feature request:**
Make it possible to define a number in the loop section. The number indicates how many times the middle section of the music is beinig played (i. e. 1 is the same as looper is turned off ). 
The looper will repeat the middle section of the music the given number of cycles and then go over the loop end mark and play the end section of the music.

**Proposal for ui:**
(dark mode) add a select box on the right beside the loop time settings. by default it set to infinite. if possible, use the infinity symbol ∞ for that.
the user can select the number of loops, e.g 2..5. if the number is set the looper behaves as described above.

**Comment by mpogue2** [comment]
I like this idea.  Another place to put the loop count might be in the timeline itself, in "marker" format (a lot of other programs use markers).

It gets a bit tricky to figure out the editing and playback, though.   In your example UI, 
I see that there's a way to set the max number of the loop (i.e. "loop 3 times"), but 
- can that be changed on the fly at playback time?  If so, does that affect the current playback loop number or the max loopback number?
- how does the user know at playback time, which number of the loop is currently playing?
- do the numbers count up or down?

Here's an example of Song Surgeon's UI:

<img width="187" alt="image" src="https://github.com/user-attachments/assets/64048092-a1a0-448f-8915-8d2eaee7ce85">

In this example, 1 is the loop number, I believe, and the 12 is the current loop number.

An interesting feature of Song Surgeon is that a loop not only defines a looping region, but the pitch, tempo, and EQ can all be changed automatically within that loop section.

I'm also thinking of a somewhat more general feature, that would allow for defining a "skip region".  I have some songs that I'd like to skip the first 7 seconds of, because the recording has a needle drop sound and 5 seconds of silence before the real music starts playing.

Overall, I like your idea!

**Comment by pkbott** [comment]
I like this too!  Can also see the value of skipping the first seven
seconds of the song, etc. I think the loop counter should go up in my
opinion so that of a glance I know how many times I’ve looped the song
because some songs are shorter than others and I may need more looping.

**Comment by Gero5** [comment]
> can that be changed on the fly at playback time? If so, does that affect the current playback loop number or the max loopback number?
> how does the user know at playback time, which number of the loop is currently playing?
> do the numbers count up or down?

Yes I think it can be changed during playback. Therefore if the planned playtime Is not enough because there was some thinks to explain I can get an extra loop when needed.
I could think of the number as being a 'no of loops left' counter. So the given number counts down every time the 'needle' returns to the loop start position.

> Song Surgeon's UI:

I guess the numbers are too small for me at club night time. :-) But nice as an information

> An interesting feature of Song Surgeon is that a loop not only defines a looping region, but the pitch, tempo, and EQ can all be changed automatically within that loop section.

Interesting feature. IMO we do not need that as callers.

> I have some songs that I'd like to skip the first 7 seconds of, because the recording has a needle drop sound and 5 seconds of silence before the real music starts playing.

Hm. I clip that in audacity. It just takes a few seconds and the problem is fixed.
